apps/04TowersOfHanoi.js

```diff
@@ -16,23 +16,39 @@ function printStacks() {
     console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-    // Your code here
+function movePiece(startStack, endStack) {
+    stacks[endStack].push(stacks[startStack].pop());
 
 }
 
-function isLegal() {
-    // Your code here
-
-}
+function isLegal(startStack, endStack) {
+    var arrayStart = stacks[startStack];
+    var arrayEnd = stacks[endStack];
+    
+    if(arrayStart[arrayStart.length - 1] < arrayEnd[arrayEnd.length -1] || arrayEnd.length === 0) {
+        return true;
+    } else {
+        return false;
+    }
+    }
 
 function checkForWin() {
     // Your code here
-
+    if (stacks.b.length === 4 || stacks.c.length === 4) {
+        console.log("You win!");
+        return true;
+    } else {
+        return false;
+    };
 }
 
+
 function towersOfHanoi(startStack, endStack) {
     // Your code here
+    isLegal(startStack, endStack);
```

Comment:
While you are correctly calling isLegal here, you are not actually using it to control whether or not to move the piece. We can use isLegal to control whether or not to move the piece using an if statement:


if(isLegal(startStack, endStack)){
//move piece
 // check for win
}